src/controllers/wishListController/addProductToWishList.ts

```diff
@@ -0,0 +1,53 @@
+import express from 'express';
+import { Request, Response } from 'express';
+import { wishListService } from '../../services/wishListService.js';
+import { UserService } from '../../services/registeruser.service.js';
+import Product from '../../models/productModel.js';
+import User from '../../models/userModel.js';
+export const addItemToWishList = async(req: Request, res: Response): Promise<void> =>{
+
+    try {
+        const userId = req.body.userId;
+        const productId = Number(req.params.productId);
+    
+        // Check if user is authenticated
+        if (!userId) {
+          res.status(401).json({Error: 'User not authenticated'});
+          return;
+        } 
+        else{
+          const user = await UserService.getUserById(userId);
+          if(user){
```

Comment:
It would be better if you create a middleware to check if the user is a buyer . In that case we can reuse it and also I did it from the changes Didas requested yesterday and now we have a middleware to check if a user is a seller  also.